lra/coordinator/server/src/main/java/io/helidon/lra/coordinator/Participant.java

```diff
@@ -0,0 +1,605 @@
+import static org.eclipse.microprofile.lra.annotation.ws.rs.LRA.LRA_HTTP_RECOVERY_HEADER;
+
+class Participant {
+
+    private static final int RETRY_CNT = 60;
+    private static final int SYNCHRONOUS_RETRY_CNT = 5;
+
+    private static final Logger LOGGER = Logger.getLogger(Participant.class.getName());
+    private final AtomicReference<CompensateStatus> compensateCalled = new AtomicReference<>(CompensateStatus.NOT_SENT);
+    private final AtomicReference<ForgetStatus> forgetCalled = new AtomicReference<>(ForgetStatus.NOT_SENT);
+    private final AtomicReference<AfterLraStatus> afterLRACalled = new AtomicReference<>(AfterLraStatus.NOT_SENT);
+    private final AtomicReference<SendingStatus> sendingStatus = new AtomicReference<>(SendingStatus.NOT_SENDING);
+    private final AtomicInteger remainingCloseAttempts = new AtomicInteger(RETRY_CNT);
+    private final AtomicInteger remainingAfterLraAttempts = new AtomicInteger(RETRY_CNT);
+
+    private final AtomicReference<Status> status = new AtomicReference<>(Status.ACTIVE);
+    private final Map<String, WebClient> webClientMap = new HashMap<>();
+    private final Map<String, URI> compensatorLinks = new HashMap<>();
+    private final long timeout;
+
+    enum Status {
+        ACTIVE(Active, null, null, false, Set.of(Completing, Compensating)),
+
+        COMPENSATED(Compensated, null, null, true, Set.of(Compensated)),
+        COMPLETED(Completed, null, null, true, Set.of(Completed)),
+        FAILED_TO_COMPENSATE(FailedToCompensate, null, null, true, Set.of()),
+        FAILED_TO_COMPLETE(FailedToComplete, null, null, true, Set.of()),
+
+        CLIENT_COMPENSATING(Compensating, COMPENSATED, FAILED_TO_COMPENSATE, false,
+                Set.of(Active, Compensated, FailedToCompensate)),
+        CLIENT_COMPLETING(Completing, COMPLETED, FAILED_TO_COMPLETE, false, Set.of(Active, Completed, FailedToComplete)),
+        COMPENSATING(Compensating, COMPENSATED, FAILED_TO_COMPENSATE, false, Set.of(Active, Compensated, FailedToCompensate)),
+        COMPLETING(Completing, COMPLETED, FAILED_TO_COMPLETE, false, Set.of(Active, Completed, FailedToComplete));
+
+        private final ParticipantStatus participantStatus;
+        private final Status successFinalStatus;
+        private final Status failedFinalStatus;
+        private final boolean finalState;
+        private final Set<ParticipantStatus> validNextStates;
+
+        Status(ParticipantStatus participantStatus,
+               Status successFinalStatus,
+               Status failedFinalStatus,
+               boolean finalState,
+               Set<ParticipantStatus> validNextStates) {
+            this.participantStatus = participantStatus;
+            this.successFinalStatus = successFinalStatus;
+            this.failedFinalStatus = failedFinalStatus;
+            this.finalState = finalState;
+            this.validNextStates = validNextStates;
+        }
+
+        public ParticipantStatus participantStatus() {
```

Comment:
class is not public, method should not be public

Reply:
fixed